shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Shelley.hs

```diff
@@ -15,3 +23,20 @@ data Shelley c
 
 instance CryptoClass.Crypto c => Era (Shelley c) where
   type Crypto (Shelley c) = c
+
+type instance Value (Shelley c) = Coin
+
+type ShelleyEra era =
```

Comment:
I still have a preference for something like `ShelleyBased`. Two reasons:

- The `Era` bit seems slightly redundant, given the type parameter is `era`.
- A user might well be confused by what the difference is between `ShelleyEra` and `Shelley`, which is an `Era`. And why e.g. `ShelleyMA` still satisfies `ShelleyEra` despite being a different era. `ShelleyBased` gives more of the sense that this derives from Shelley.

However, this is pure bikeshedding, so if I haven't convinced you then feel free to leave it as is :-)

Reply:
I totally agree with @nc6

Reply:
I know naming is hard, and I'm not good in it either, but I now realise that I *really* don't want `ShelleyEra` to be used as the name for this.

This module defines `data Shelley c` as the tag for the Shelley era. Since that is already rather confusing in consensus, we use the `ShelleyEra` type synonym (`type ShelleyEra c = Era.Shelley c`) for it.

Now `ShelleyEra` is also going to be defined here as a type synonym for a bunch of constraints that are not even specific to the Shelley era (they're parametric in `era`)? :exploding_head: 

Since these constraints are parametric in `era` and not specific to Shelley, why define them in this module (named `Cardano.Ledger.Shelley`)? I think they should be put in a non-era specific module.

Also, why not just add them as superclass constraints of the `Era` class? Maybe some of them don't belong there -- oh, I just realised that `Val (Value era)` is in the list (`Val` is also not the best name, what about `IsValue`? Or use `Value` for the class and `EraValue era` for the type family?), which used to have a bunch of these constraint already, so they're just being moved from one place to another. What was the reason for removing them from `Val` (maybe there's a good one I don't see as an outsider)? From my point of view, having them as superclass constraints of `Val` makes much more sense (maybe not the `CompactForm` ones?).

Reply:
@mrBliss I argued against having them on `Val` or `Era` because of a few reasons:
- It's helpful to see where we actually need rather than carrying them around everywhere.
- Having them on `Era` would imply a universal quantification over all eras, which is a pretty strong condition. Whereas what we really have is a set of constraints on eras which are sufficiently similar to Shelley. Maybe these will end up identical, IDK, but from a conceptual viewpoint I much prefer not having them on era.
- `Val` is a class with semantics, where the superclasses help to define those semantics. The additional constraints have nothing to do with what it means to be a `Val`, they were just recorded there for convenience. Personally I really dislike such aggregation of constraints.

Reply:
I see, that sounds reasonable. My main gripe is with calling it `ShelleyEra`. Calling it `ShelleyBased` would solve that. :+1: